### PR TITLE
test: end point to get intents across all zones for a delegator address #466

### PR DESCRIPTION
**Pull request by muku314115**
## 1. Summary
Fixes # (issue)
Pulled into a new branch and recreated the PR due to merge problems in the previous branch
Built on top of https://github.com/ingenuity-build/quicksilver/pull/465

## 2.Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## 5. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

**Review by joe-bowman — DISMISSED** [review]
lgtm

File: x/interchainstaking/keeper/grpc_query.go [hunk 1]
```diff
@@ -146,6 +146,26 @@ func (k *Keeper) DelegatorIntent(c context.Context, req *types.QueryDelegatorInt
 	return &types.QueryDelegatorIntentResponse{Intent: &intent}, nil
 }
 
+// DelegatorIntents returns information about the delegation intent of the caller for all zones.
```
Comment (aljo242) [inline x/interchainstaking/keeper/grpc_query.go:149]:
```suggestion
// DelegatorIntents returns information about the delegation intent of the given delegator for all zones.
```

File: x/interchainstaking/keeper/grpc_query.go [hunk 2]
```diff
@@ -146,6 +146,26 @@ func (k *Keeper) DelegatorIntent(c context.Context, req *types.QueryDelegatorInt
...
+
+	ctx := sdk.UnwrapSDKContext(c)
+
+	intents := []*types.DelegatorIntentsResponse{}
+
+	k.IterateZones(ctx, func(_ int64, zone *types.Zone) bool {
+		intent, _ := k.GetDelegatorIntent(ctx, zone, req.DelegatorAddress, false)
+
+		intents = append(intents, &types.DelegatorIntentsResponse{ChainId: zone.ChainId, Intent: &intent})
+		return false
```
Comment (aljo242) [inline x/interchainstaking/keeper/grpc_query.go:163]: I think we should not append in the cases where `intent` is empty.  Here, we are appending for all zones, whether or not there is any info.

Comment (muku314115) [reply x/interchainstaking/keeper/grpc_query.go]: added a nil check to ensure we don't append empty intents

**Comment by joe-bowman** [comment]
Other than a `make proto-format` looks good to go.

**Review by joe-bowman — APPROVED** [review]
lgtm

**Review by ajansari95 — APPROVED** [review]
lgtm

**Pull request merged**